File: java/src/main/java/com/techelevator/service/OrderService.java
package com.techelevator.service;

import com.techelevator.dao.JdbcOrderDao;
import com.techelevator.dao.OrderDao;
import com.techelevator.model.Order;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class OrderService {
    OrderDao orderDao;

    public OrderService(OrderDao orderDao){
        this.orderDao = orderDao;
    }

    public int placeOrder(Order order){
        return orderDao.placeOrder(order);
    }

    public Order getOrderById(int id){
        return orderDao.getOrderById(id);
    }

    public List<Order> getListOfOrders(String status){
        if(status == null || status.equals("")) {
            return orderDao.getListOfOrdersInProcess();
        } else {
            return orderDao.getListOfOrdersByStatus(status);
        }
    }
}

File: vue/src/services/OrderService.js
import axios from 'axios';

export default {

    getOrderByID(id){
        return axios.get(`/order/${id}`);
    },
    getAllOrders(){
        return axios.get('/order');
    },
    placeOrder(order){
        return axios.post('/order', order);
    },
    getOrdersByStatus(status){
        return axios.get(`/order?status=${status}`)
    },
    updateOrderStatus(id, status){
        return axios.put(`/order/${id}/status`, {status})
    }
}